tests/test_invenio_access_permissions.py

```diff
@@ -82,6 +82,41 @@ def test_invenio_access_permission(app):
         assert permission.allows(fake_identity)
 
 
+def test_invenio_access_permission_denied(app):
+    """Permission is always denied when excluded."""
+    fake_identity = FakeIdentity()
+
+    InvenioAccess(app)
+    with app.test_request_context():
+        db.session.begin(nested=True)
+        # Create two users
+        user_one = User(email='test@inveniosoftware.org')
+        user_two = User(email='test2@inveniosoftware.org')
+
+        # Create permission
+        permission = Permission(ActionNeed('read'))
+        # Exclude user one
+        permission.explicit_excludes.add(UserNeed(user_two.id))
+
+        # Check the correct permissions are granted/excluded
+        assert not permission.allows(fake_identity)
+
+        db.session.add(ActionUsers(action='read', user=user_one))
+        db.session.add(ActionUsers(action='read', user=user_two))
+        db.session.commit()
+
+        assert not permission.allows(fake_identity)
```

Comment:
if I may suggest, but not sure if it is right... it is just that it took me sometime to understand...

```
        # Create two users
        user_one = User(email='test@inveniosoftware.org')
        user_two = User(email='test2@inveniosoftware.org')
        
        # Create permission
        permission = Permission(ActionNeed('read'))

        # Allow user one
        permission.explicit_needs.add(UserNeed(user_one.id))
        # Forbid user two
        permission.explicit_excludes.add(UserNeed(user_two.id))
        ...

        # identity without user needs is allowed
        assert not permission.allows(fake_identity)

        # identity with wrong user needs is not allowed
        fake_identity.provides.add(UserNeed(user_one.id))
        assert not permission.allows(fake_identity)

        # identity with right user needs is not allowed
        fake_identity.provides.add(UserNeed(user_two.id))
        assert permission.allows(fake_identity)
```

Reply:
Fixed this :) :rocket: